panos_upgrade_assurance/firewall_proxy.py

```diff
@@ -1402,3 +1402,61 @@ def get_device_software_version(self):
         self._fw.get_device_version()
         fw_version = self._fw.version.replace("-h", ".")
         return version.parse(fw_version)
+
+    def get_fib(self) -> dict:
+        """
+        Get the information from the forwarding information table (FIB).
+
+        The actual API command run is `show routing fib`.
+
+        # Returns
+
+        dict: Status of the route entries in the FIB
+        ```python showLineNumbers title="Sample output"
+        {
+            '0.0.0.0/0_ethernet1/1': {
+                'Destination': '0.0.0.0/0',
+                'Interface': 'ethernet1/1',
+                'Next Hop Type': '0',
+                'Flags': 'ug',
+                'Next Hop': '10.10.11.1',
+                'MTU': '1500'
+            },
+            '1.1.1.1/32_loopback.10': {
+                'Destination': '1.1.1.1/32',
+                'Interface': 'loopback.10',
+                'Next Hop Type': '3',
+                'Flags': 'uh',
+                'Next Hop': '0.0.0.0',
+                'MTU': '1500'
+            }
+        }
+        ```
+
+        """
+        response = self.op_parser(cmd="show routing fib")
+        
+        fibs = response["fibs"]["entry"]
+        
+        results = {}
+        
+        for fib_entry in fibs:
```

Comment:
make sure you are not iterating over a `dict` instead of a map. See comment below.

A common workaround is either to iterate over XML, without parsing to `dict` or using something like: `for fib in fibs if isinstance(fibs, list) else [fibs]:`

Reply:
Fixed this during the session with @FoSix .